src/manipulation.js

```diff
@@ -193,14 +193,19 @@ function domManip( collection, args, callback, ignored ) {
 						!dataPriv.access( node, "globalEval" ) &&
 						jQuery.contains( doc, node ) ) {
 
-						if ( node.src ) {
+						if ( node.src && node.type !== "module" ) {
```

Comment:
It's very convenient that module execution is inherently asynchronous with respect to parsing. But I suppose we'll also need awareness of [`nomodule`](https://html.spec.whatwg.org/#attr-script-nomodule) now.